tests/ui/typeck/suggestions/suggest-clone-in-macro-issue-139253.rs

```diff
@@ -0,0 +1,21 @@
+// Make sure we don't suggest clone not in dbg macro
+
+#[derive(Clone, Debug)]
+struct Point<T> {
+    v: T,
+}
+
+#[derive(Clone, Debug)]
+struct S;
+
+fn foo(_: S) {}
+
+fn test1() {
+    let s = &S;
+    foo(dbg!(s)); //~ ERROR mismatched types
+}
+
+fn main() {
+    let a: Point<u8> = dbg!(Point { v: 42 });
+    let b: Point<u8> = dbg!(&a); //~ ERROR mismatched types [E0308]
```

Comment:
Since the `Point` test case is basically identical to the `S` test case, I would just remove the former in favor the latter.